### DISCUSSION TITLE
Trying to use #[server]

### DISCUSSION THREAD

**Discussion by tiswas247** [DioxusLabs/dioxus#3126] [General]
The code below users a server function in dioxus 0.5.6.

I want to get the `Vec<Model>` and iterate through it.

Currently the `use_server_future` appears to return `ReadOnlySignal<Option<Result<Vec<Model>, ServerFnError>>>`.

How do I get into that and get my models?

Also, if I add a parameter to the get_models i.e. get_models(id: i32) what would be the syntax to call the method?

The pattern I want to follow is to load data from the server on each page load and fully populate the page.

Thanks

```rust
#![allow(non_snake_case)]

use dioxus::prelude::*;
use dioxus_logger::tracing;
use serde::{Deserialize, Serialize};

#[derive(Clone, Routable, Debug, PartialEq, serde::Serialize, serde::Deserialize)]
enum Route {
    #[route("/")]
    Home {},
}

fn main() {
    // Init logger
    dioxus_logger::init(tracing::Level::INFO).expect("failed to init logger");
    tracing::info!("starting app");
    launch(App);
}

fn App() -> Element {
    rsx! {
        Router::<Route> {}
    }
}

#[component]
fn Home() -> Element {
    // Get the signal from use_server_future
    let models = use_server_future(get_models)?.value();

    rsx! {
        "{models:?}"
    }
}

#[derive(Serialize, Deserialize, Debug)]
pub struct Model {
    name: String,
    base_url: String
}

#[server(GetModels)]
async fn get_models() -> Result<Vec<Model>, ServerFnError> {
    Ok(
        vec![
            Model {
                name: "Model 1".into(),
                base_url: "http://".into()
            },
            Model {
                name: "Model 2".into(),
                base_url: "http://".into()
            }
        ]
    )
}
```

**Comment by wiseaidev** [comment]
You can use the [`use_resource`](https://dioxuslabs.com/learn/0.5/reference/use_resource) hook for this. Have a look at [this detailed example](https://github.com/opensass/aibook/blob/93a5aa76bd8acbb693d867a33b9876aedc025d5a/src/components/dashboard/books/list.rs#L32-L64) calling a server function with params [`get_books_for_user`](https://github.com/opensass/aibook/blob/93a5aa76bd8acbb693d867a33b9876aedc025d5a/src/server/book/controller.rs#L167).
